pool/src/pool/config.rs

```diff
@@ -206,7 +264,82 @@ mod tests {
             execute_update_pool(&e, 1_000_000_000u64);
         });
     }
+    #[test]
+    fn test_execute_queued_initialize_reserve() {
+        let e = Env::default();
+        let pool = testutils::create_pool(&e);
+        let bombadil = Address::generate(&e);
 
+        let (asset_id_0, _) = testutils::create_token_contract(&e, &bombadil);
+
+        let metadata = ReserveConfig {
+            index: 0,
+            decimals: 7,
+            c_factor: 0_7500000,
+            l_factor: 0_7500000,
+            util: 0_5000000,
+            max_util: 0_9500000,
+            r_one: 0_0500000,
+            r_two: 0_5000000,
+            r_three: 1_5000000,
+            reactivity: 100,
+        };
+        e.as_contract(&pool, || {
+            storage::set_queued_reserve_init(
+                &e,
+                &QueuedReserveInit {
+                    new_config: metadata.clone(),
+                    unlock_block: e.ledger().sequence(),
+                },
+                &asset_id_0,
+            );
+            execute_queued_reserve_initialization(&e, &asset_id_0);
+            let res_config_0 = storage::get_res_config(&e, &asset_id_0);
+            assert_eq!(res_config_0.decimals, metadata.decimals);
+            assert_eq!(res_config_0.c_factor, metadata.c_factor);
+            assert_eq!(res_config_0.l_factor, metadata.l_factor);
+            assert_eq!(res_config_0.util, metadata.util);
+            assert_eq!(res_config_0.max_util, metadata.max_util);
+            assert_eq!(res_config_0.r_one, metadata.r_one);
+            assert_eq!(res_config_0.r_two, metadata.r_two);
+            assert_eq!(res_config_0.r_three, metadata.r_three);
+            assert_eq!(res_config_0.reactivity, metadata.reactivity);
+            assert_eq!(res_config_0.index, 0);
+        });
+    }
+    #[test]
+    #[should_panic(expected = "Error(Contract, #7)")]
+    fn test_execute_queued_initialize_reserve_requires_block_passed() {
+        let e = Env::default();
+        let pool = testutils::create_pool(&e);
+        let bombadil = Address::generate(&e);
+
+        let (asset_id_0, _) = testutils::create_token_contract(&e, &bombadil);
+
+        let metadata = ReserveConfig {
+            index: 0,
+            decimals: 7,
+            c_factor: 0_7500000,
+            l_factor: 0_7500000,
+            util: 0_5000000,
+            max_util: 0_9500000,
+            r_one: 0_0500000,
+            r_two: 0_5000000,
+            r_three: 1_5000000,
+            reactivity: 100,
+        };
+        e.as_contract(&pool, || {
+            storage::set_queued_reserve_init(
+                &e,
+                &QueuedReserveInit {
+                    new_config: metadata.clone(),
+                    unlock_block: e.ledger().sequence() + 1,
+                },
+                &asset_id_0,
+            );
+            execute_queued_reserve_initialization(&e, &asset_id_0);
+        });
+    }
```

Comment:
need a test for `cancel` if we deem it should be kept

Reply:
test_execute_cancel_queued_reserve_initialization